streams/src/test/java/org/apache/kafka/streams/integration/TaskMetadataIntegrationTest.java

```diff
@@ -158,9 +158,13 @@ public void shouldReportCorrectEndOffsetInformation() {
         }
     }
 
-    private TaskMetadata getTaskMetadata(final KafkaStreams kafkaStreams) {
+    private TaskMetadata getTaskMetadata(final KafkaStreams kafkaStreams) throws InterruptedException {
+        TestUtils.waitForCondition( () -> kafkaStreams
```

Comment:
I'm sure the odds of this are super low, but to really make this airtight you'd need to have the `List<TaskMetadata>` that you use for the test be the same one that you test in this condition, otherwise it's _possible_ for the one task to have been revoked in the split second between `waitForCondition()` and the new call to `metadataForLocalThreads()`.

Reply:
Correct me if I'm wrong, I just want to understand better the case you are raising. You mean to keep the `TaskMetadata` used for this test and then compare them with the `activeTasks`, right?
But I don't really understand how this will prevent from discrepancies if the task has been revoked in between calls. Wouldn't it be then also not present in `activeTasks`?

Sorry to hijack this for this question and thanks in advance!

Reply:
@ableegoldman good point. I was trying to avoid referencing a variable inside a lambda but I can just use an atomic reference

Reply:
@jlprat We are actually getting the `TaskMetadata` for that active tasks here and then   making sure that metadata is as expected.

Reply:
Got it, thanks!